service/frontend/adminHandler.go

```diff
@@ -1810,3 +1820,30 @@ func convertFilterListToMap(filters []*types.DynamicConfigFilter) (map[dc.Filter
 	}
 	return newFilters, nil
 }
+
+// Ensures that shards received are within the range of the server,
+// or they'll be filtered out. This necessarily means that replication requests
+// that are for shards that don't exist are dropped.
```

Comment:
if the replication is fundamentally can't be done completely in such setups, should we even attempt to do the partial replication?

Reply:
it can be - as far as I can tell, successfully be done when scaling a cluster up, in terms of the number of shards. Just not down. This strikes me [as useful](https://uber-cadence.slack.com/archives/CL22WDF70/p1703179266189009).